lib/stdlib/test/property_test/queue_prop.erl

```diff
@@ -0,0 +1,670 @@
+%%
+%% %CopyrightBegin%
+%%
+%% Copyright Ericsson AB 2021. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%
+%% %CopyrightEnd%
+%%
+-module(queue_prop).
+
+-compile(export_all).
+
+-proptest(eqc).
+-proptest([triq, proper]).
+
+-ifndef(EQC).
+-ifndef(PROPER).
+-ifndef(TRIQ).
+-define(EQC, true).
+-endif.
+-endif.
+-endif.
+
+-ifdef(EQC).
+-include_lib("eqc/include/eqc.hrl").
+-define(MOD_eqc,eqc).
+
+-else.
+-ifdef(PROPER).
+-include_lib("proper/include/proper.hrl").
+-define(MOD_eqc,proper).
+
+-else.
+-ifdef(TRIQ).
+-define(MOD_eqc,triq).
+-include_lib("triq/include/triq.hrl").
+
+-endif.
+-endif.
+-endif.
+
+%%%%%%%%%%%%%%%%%%
+%%% Properties %%%
+%%%%%%%%%%%%%%%%%%
+
+prop_new() ->
+    [] =:= queue:to_list(queue:new()).
+
+prop_is_queue() ->
+    ?FORALL(
+	{IsQueue, Q},
+	oneof([
+	    {true, queue()},
+	    {false, non_queue()}
+	]),
+	begin
+	    IsQueue =:= queue:is_queue(Q)
+	end
+    ).
+
+prop_list_conversion() ->
+    ?FORALL(
+	List,
+	list(),
+	begin
+	    Queue = queue:from_list(List),
+	    queue:is_queue(Queue) andalso
+	    List =:= queue:to_list(Queue)
+	end
+    ).
+
+prop_all() ->
+    ?FORALL(
+	{L, Q},
+	oneof([list_queue(atom()), list_queue(term())]),
+	begin
+	    lists:all(fun is_atom/1, L) =:= queue:all(fun is_atom/1, Q)
+	end
+    ).
+
+prop_any() ->
+    ?FORALL(
+	{L, Q},
+	list_queue(),
+	begin
+	    lists:any(fun is_atom/1, L) =:= queue:any(fun is_atom/1, Q)
+	end
+    ).
+
+prop_cons() ->
+    common_in_r_cons(cons).
+
+prop_daeh() ->
+    common_get_r_last_daeh(daeh).
+
+prop_delete() ->
+    ?FORALL(
+	{X, {L, Q}},
+	{term(), list_queue()},
+	begin
+	    R1 = if
+		L =:= [] ->
+		    true;
+		true ->
+		    {_, Y} = hd(lists:sort([{rand:uniform(), I} || I <- L])),
```

Comment:
This is a rather slow way of picking a random element from a list.  Why not `lists:nth(rand:uniform(length(L)), L)`?

Reply:
Yeah... what was I thinking back then, I wonder... :sweat_smile: I'll change it as suggested.